spec/requests/trains_request_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe "Trains" do
+  describe "GET /index" do
+    it "returns http success" do
+      get trains_path
+      expect(response).to be_successful
```

Comment:
not checking the response body for train data?
Also your not stubbing out the request in tests, which means its actually trying to make that request which it shouldnt be